### PR TITLE
Replace `oColorsGetUseCase` with `oColorsByUsecase` and related changes.

### PR DESCRIPTION
**Pull request by notlee**
- Replace `oColorsGetUseCase` with `oColorsByUsecase`, which has
updated arguments to support explicit namespacing and other
proposed features.
- `oColorsByUsecase` does not accept `all` as a property. If a
usecase has an `all` colour set, that will be used for any given
property which hasn't been assigned a colour explicitly.
- Usecase CSS classes are no longer output, and the related option
is removed from `oColors`.
- `oColorsByUsecase` throws an error when a usecase property is
missing unless a fallback has been given.

Relates to: https://github.com/Financial-Times/o-colors/issues/198

### CODE REVIEW

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to 92338d9da82e17111b0a0c5febfb7d910d2bf106
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

File: MIGRATION.md [hunk 1]
```diff
@@ -2,19 +2,96 @@
...
+@include oColors($opts: (
+	'palette-custom-properties': true,
+	'palette-classes': true,
+-	'usecase-classes': true
+));
+```
+
+#### oColorsByUsecase
+
+`oColorsGetUseCase` is now `oColorsByUsecase`. By default `oColorsByUsecase` it now errors if a usecase isn't found, unless a `$fallback` colour has been given (which may be `null`). A list of usecases is no longer accepted. `oColorsByUsecase` also has a new `$from` argument, to get usecases set by different component or projects.
```
Comment (JakeChampion) [inline MIGRATION.md]:
```suggestion
`oColorsGetUseCase` is now `oColorsByUsecase`. By default `oColorsByUsecase` now errors if a usecase isn't found, unless a `$fallback` colour has been given (which may be `null`). A list of usecases are no longer accepted. `oColorsByUsecase` also has a new `$from` argument, to get usecases set by different component or projects.
```

File: MIGRATION.md [hunk 2]
```diff
@@ -2,19 +2,96 @@
...
+
+If getting a usecase set by o-colors update the mixin name:
+```diff
+-background-color: oColorsGetUseCase('page', 'background');
++background-color: oColorsByUsecase('page', 'background');
+```
+
+Not `all` is no longer a valid property when getting a usecase. Instead use one of 'text', 'background', 'border', or 'outline'.
```
Comment (JakeChampion) [inline MIGRATION.md]:
```suggestion
Note `all` is no longer a valid property when getting a usecase. Instead use one of 'text', 'background', 'border', or 'outline'.
```

File: README.md [hunk 3]
```diff
@@ -233,6 +206,28 @@ section-house-home-alt |all
 section-money |         all
 section-money-alt |     all
 
+## Defining custom use cases
+
+You can add use cases for your particular component or product. This is done using the `oColorsSetUseCase` mixin:
```
Comment (JakeChampion) [inline README.md]:
```suggestion
You can add usecases for your particular component or product. This is done using the `oColorsSetUseCase` mixin:
```

File: README.md [hunk 4]
```diff
@@ -233,6 +206,28 @@ section-house-home-alt |all
 section-money |         all
 section-money-alt |     all
 
+## Defining custom use cases
```
Comment (JakeChampion) [inline README.md]:
```suggestion
## Defining custom usecases
```

File: README.md [hunk 5]
```diff
@@ -233,6 +206,28 @@ section-house-home-alt |all
...
+
+You can add use cases for your particular component or product. This is done using the `oColorsSetUseCase` mixin:
+
+```scss
+@include oColorsSetUseCase(email, text, 'black-60');
+```
+
+It takes three arguments:
+
+* **Use case**: your particular use case
```
Comment (JakeChampion) [inline README.md]:
```suggestion
* **Usecase**: your particular usecase
```

File: README.md [hunk 6]
```diff
@@ -233,6 +206,28 @@ section-house-home-alt |all
...
+* **Property**: the property for which the color should be used for (background, border, or text)
+* **Color**: a color from the palette
+
+If you are creating a use case for a component, you *must* namespace your use case name with the name of your component.
```
Comment (JakeChampion) [inline README.md]:
```suggestion
If you are creating a usecase for a component, you *must* namespace your usecase name with the name of your component.
```

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to a6166f2a32f052e8ee805c65a51f2084a033be41
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to adfac8d58353f7f7d3808223ae1c70ddec96bcbc
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to e23120525baf0340f28558ece89d337ce429ee5f
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to c76bba9813d9fca98be6d51cc395aedeccb9cd5b
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to 0dfcce530d48593dcf7f55d3c90ee842163e5b30
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

**Comment by origamiserviceuser** [comment]
o-colors bundle size difference from 4.10.3 to 7bddf12d88cecd965928de4298a2c831c375be3e
css, master: 1.62kb decrease (-0.31kb/gzip)
css, internal: 0.52kb decrease (-0.12kb/gzip)
css, whitelabel: 0.21kb decrease (-0.05kb/gzip)

**Comment by notlee** [comment]
Thanks Fox friend, it's only a README though friend.

**Pull request merged**